src/venstarcolortouch/venstarcolortouch.py

```diff
@@ -109,7 +109,8 @@ def login(self):
         j = r.json()
         if j["api_ver"] >= MIN_API_VER:
             self._api_ver = j["api_ver"]
-            self._firmware_ver = tuple(map(int, j["firmware"].split(".")))
+            if "firmware" in j:
```

Comment:
I think you need an 'else:' here because if firmware is not in j, then it will retain the pre-set of 'None' which can not be compared using '<' later.

Maybe:
`else:
    self._firmware_ver = (0,0)
`